pallets/transfer-allowlist/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+		_,
+		Twox64Concat,
+		AccountIdOf<T>,
+		Twox64Concat,
+		CurrencyIdOf<T>,
+		u64,
+		OptionQuery,
+	>;
+
+	/// Storage item for allowances specified for a sending account, currency type and recieving location
+	#[pallet::storage]
+	#[pallet::getter(fn sender_currency_reciever_allowance)]
+	pub type AccountCurrencyTransferAllowance<T> = StorageNMap<
+		_,
+		(
+			NMapKey<Twox64Concat, AccountIdOf<T>>,
+			NMapKey<Twox64Concat, CurrencyIdOf<T>>,
+			NMapKey<Blake2_128Concat, Location>,
+		),
+		AllowanceDetails<BlockNumberOf<T>>,
+		OptionQuery,
+	>;
+	/// Storage item for Allowance delays for a sending account/currency
+	#[pallet::storage]
+	#[pallet::getter(fn sender_currency_delay)]
+	pub type AccountCurrencyDelay<T> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		AccountIdOf<T>,
+		Twox64Concat,
+		CurrencyIdOf<T>,
+		BlockNumberOf<T>,
+		OptionQuery,
+	>;
+
+	//
+	// Pallet Errors and Events
+	//
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Number of allowances for a sending Account/Currency set at max allowance count storage type val (currently u64::MAX)
+		AllowanceCountOverflow,
+		/// An operation expecting one or more allowances for a sending Account/Currency set, where none present
+		NoAllowancesSet,
+		/// Attempted to create allowance for existing Sending Account, Currency, and Receiver combination
+		ConflictingAllowanceSet,
+		/// CatchAll for Allowance Count arithmetic errors -- largely for coverage for errors that should be impossible
+		AllowanceCountArithmeticError,
+		/// No matching allowance for Location/Currency
+		NoMatchingAllowance,
+		/// No matching delay
+		NoMatchingDelay,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Event for successful creation of a transfer allowance
+		TransferAllowanceCreated {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+			allowed_at: BlockNumberOf<T>,
+			blocked_at: BlockNumberOf<T>,
+		},
+		/// Event for successful removal of transfer allowance perms
+		TransferAllowanceRemoved {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+			allowed_at: BlockNumberOf<T>,
+			blocked_at: BlockNumberOf<T>,
+		},
+		/// Event for successful removal of transfer allowance perms
+		TransferAllowancePurged {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+		},
+		/// Event for Allowance delay update
+		TransferAllowanceDelaySet {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+			delay: BlockNumberOf<T>,
+		},
+		/// Event for Allowance delay removal
+		TransferAllowanceDelayRemoval {
+			sender_account_id: AccountIdOf<T>,
+			currency_id: CurrencyIdOf<T>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T>
+	where
+		<T as frame_system::Config>::AccountId: Into<Location>,
+	{
+		#[transactional]
+		#[pallet::call_index(0)]
```

Comment:
Do we need this line?

Reply:
Ah, yeah given that we've gotten the conversions taking place before this with from/into, this would not be necessary.  Will remove.

Reply:
👍 removed